cpp/include/cudf/utilities/traits.hpp

```diff
@@ -42,6 +43,15 @@ struct is_relationally_comparable_impl<
     L, R, void_t<less_comparable<L, R>, greater_comparable<L, R> > >
     : std::true_type {};
 
+template <typename T>
+using is_timestamp_t = simt::std::disjunction<
+  std::is_same<cudf::timestamp_D, T>,
+  std::is_same<cudf::timestamp_s, T>,
+  std::is_same<cudf::timestamp_ms, T>,
+  std::is_same<cudf::timestamp_us, T>,
+  std::is_same<cudf::timestamp_ns, T>
+>;
```

Comment:
Excited to use the new [C++17 SFINAE types](https://en.cppreference.com/w/cpp/types/disjunction) via `libcudacxx`.